c++11/src/tracer.cpp

```diff
@@ -66,10 +86,60 @@ LightStepTracer::MakeSpanContext(
 }
 
 //------------------------------------------------------------------------------
+// MakeThreadedTracer
+//------------------------------------------------------------------------------
+static std::shared_ptr<LightStepTracer> MakeThreadedTracer(
+    std::shared_ptr<spdlog::logger> logger, LightStepTracerOptions&& options) {
+  std::unique_ptr<SyncTransporter> transporter;
+  if (options.transporter != nullptr) {
+    transporter = std::unique_ptr<SyncTransporter>{
+        dynamic_cast<SyncTransporter*>(options.transporter.get())};
+    if (transporter == nullptr) {
+      logger->error(
+          "`options.transporter` must be derived from SyncTransporter");
+      return nullptr;
+    }
+    options.transporter.release();
+  } else {
+    transporter = MakeGrpcTransporter(*logger, options);
+  }
+  auto recorder = std::unique_ptr<Recorder>{
+      new AutoRecorder{*logger, std::move(options), std::move(transporter)}};
+  return std::shared_ptr<LightStepTracer>{
+      new LightStepTracerImpl{std::move(logger), std::move(recorder)}};
+}
+
+//------------------------------------------------------------------------------
+// MakeNonThreadedTracer
```

Comment:
nit: I'd prefer to call this MakeSingleThreadedTracer. any reason that's less clear?

Reply:
That's better. I'll change it.